src/mpi/comm/comm_split_type.c

```diff
@@ -607,6 +607,93 @@ static int network_split_by_minsize(MPIR_Comm * comm_ptr, int key, int subcomm_m
   fn_fail:
     goto fn_exit;
 }
+
+static int network_split_by_min_memsize(MPIR_Comm * comm_ptr, int key, long min_mem_size,
+                                        MPIR_Comm ** newcomm_ptr)
+{
+
+    int mpi_errno = MPI_SUCCESS;
+    int i, color;
+    netloc_node_t *network_node;
+    int comm_size = MPIR_Comm_size(comm_ptr);
+
+    /* Get available memory in the node */
+    hwloc_obj_t memory_obj = NULL;
+    long total_memory_size = 0;
+    long *memory_at_nodes = NULL;
+
+    while ((memory_obj =
+            hwloc_get_next_obj_by_type(MPIR_Process.hwloc_topology, HWLOC_OBJ_NUMANODE,
+                                       memory_obj)) != NULL) {
+        total_memory_size += memory_obj->total_memory;
+    }
+
+    if (min_mem_size == 0 || comm_size < min_mem_size ||
+        MPIR_Process.network_attr.type == MPIR_NETLOC_NETWORK_TYPE__INVALID) {
+        color = MPI_UNDEFINED;
+    } else
+        if (MPIR_Process.network_attr.type == MPIR_NETLOC_NETWORK_TYPE__FAT_TREE ||
```

Comment:
`else if` block should go on the same line.

Reply:
This was the whitespace checker, I think. I'll make the change and run the check one more time to confirm my suspicion.